File: 645-SetMismatch/645-SetMismatch.py
# Last updated: 07/04/2026, 11:38:21
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        hash = {}
        res= []
        for i in nums :
            if i in hash :
                hash[i] = 1
                if hash[i] == 1 :
                    res.append(i)
            else:
                hash[i] = 0 


        n = len(nums)
        j = ((n*(n+1))//2 ) - sum(set(nums))
        res.append(j)
        return res


            

            

            
            


        

File: 645-SetMismatch/645-SetMismatch.cpp
// Last updated: 30/04/2026, 16:19:27
1class Solution {
2public:
3    vector<int> findErrorNums(vector<int>& nums) {
4
5        int n = nums.size();
6        vector<int> a (n+1,-1);
7        int r= -1;
8
9        for ( int i = 0 ; i <n;i++){
10            if(a[nums[i]] != -1)  r = nums[i];
11            if ( a[nums[i]] == -1) a[nums[i]] = nums[i];
12        }
13        int m = -1;
14        for ( int i = 1; i<=n;i++){
15            if ( a[i] == -1){
16                m = i;
17                break;
18            }
19        }
20        return {r,m};
21
22    }
23};